progress/copycallback.go

```diff
@@ -7,26 +7,30 @@ import (
 
 type CopyCallback func(totalSize int64, readSoFar int64, readSinceLast int) error
 
-type bodyWithCallback struct {
+type BodyWithCallback struct {
 	c         CopyCallback
 	totalSize int64
 	readSize  int64
 	ReadSeekCloser
 }
 
-func NewByteBodyWithCallback(by []byte, totalSize int64, cb CopyCallback) ReadSeekCloser {
+var _ ReadSeekCloser = (*BodyWithCallback)(nil)
```

Comment:
These declarations are unnecessary. It wouldn't compile if this wasn't the case.